proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+
+**When to use filesystem storage:**
+
+- Deployments where eliminating object storage dependency is preferred
+- Environments where Kubeflow distributions or platform providers prefer not to support additional storage systems
+- Multi-tenant deployments requiring per-namespace storage isolation, scaling, and quotas
+- Development and experimentation scenarios
+
+**When NOT to use filesystem storage:**
+
+- High-throughput production workloads with large artifacts
+- Scenarios requiring object storage features (versioning, lifecycle policies, geo-replication)
+- When existing object storage infrastructure is already available and preferred
+
+### Goals
+
+Based on the user story "As a user, I want to provision Kubeflow Pipelines with just a PVC for artifact storage so that I can quickly get started", this KEP aims to:
+
+1. **Add filesystem storage as an additional backend option** alongside S3-compatible and Google Cloud Storage, primarily using PVC but not limited to it
+2. **Enable zero-configuration storage** for experimentation use cases - a KFP server can be installed with just a PVC for artifact storage
+3. **Provide namespace-isolated artifact storage** with proper subject access review guards in multi-user mode
+4. **Allow any Kubernetes access mode to be configured** - KFP passes through the configuration to Kubernetes (RWO default)
+5. **Support existing pipelines** that use KFP's standard artifact types (Dataset, Model, etc.) - pipelines work unchanged with the new filesystem backend
+6. **Match existing artifact persistence behavior** - artifacts persist indefinitely until explicitly deleted (no automatic cleanup)
+7. **Enable separate scaling of artifact serving** through an artifacts-only KFP instance with `--artifacts-only` flag
+8. **Ensure UI can properly handle artifact downloads** with the new `kfp-artifacts://` URI scheme
+
+### Non-Goals
+
+1. **Replace object storage** - filesystem storage is an additional option, not a replacement
+2. **Cross-namespace artifact sharing** - Initial implementation focuses on namespace isolation
+3. **Performance optimization for large-scale workloads** - filesystem storage may not match object storage performance for massive datasets or high concurrency scenarios
+4. **Storage migration tools** - Migrating from S3 to PVC/filesystem is not supported
+5. **Advanced storage features** - No versioning, replication, or geo-distribution initially
+
+## Proposal
+
+### Overview
+
+This KEP proposes adding a new artifact storage backend that uses filesystem storage (primarily Kubernetes `PersistentVolumeClaims`) instead of object storage. The implementation will:
+
+1. Create one PVC per namespace for artifact storage
+2. Use configurable access mode with sensible defaults (RWO)
+3. Organize artifacts in a filesystem hierarchy within the PVC
+4. Provide transparent access through the existing KFP artifact APIs with new `kfp-artifacts://` URI scheme
```

Comment:
Do we have "KFP artifact APIs"? Perhaps this is referencing to @HumairAK 's MLMD removal PR.